File: Introduction/TwoSets.java

import java.util.Scanner;
import java.util.Stack;

public class TwoSets {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        if(((n*(n+1))/2)%2== 0){
            System.out.println("YES");
            long a, b; a = b= 0;
            Stack<Integer> s1 = new Stack<>();
            Stack<Integer> s2 = new Stack<>();
            while(n > 0){
                if(a <= b){
                    a+=n;
                    s1.add(n);
                }else{
                    b+=n;
                    s2.add(n);
                }
                n--;
            }
            System.out.println();
            System.out.println(s1.size());
            while(!s1.isEmpty()){
                System.out.print(s1.pop() + " ");
            }
            System.out.println();
            System.out.println(s2.size());
            while(!s2.isEmpty()){
                System.out.print(s2.pop() + " ");
            }
        }else{
            System.out.println("NO");
        }
        sc.close();
    }
}

File: Introduction/TwoSets.cpp
#include <iostream>
#include <stack>
using namespace std;
int main(){
    int n;
    cin >> n;   
    if(((n*(n+1))/2)%2== 0){
            cout << "YES" << endl;
            stack<int> s1;
            stack<int> s2;
            long a, b; a = b= 0;
            while(n > 0){
                if(a <= b){
                    a+=n;
                    s1.push(n);
                }else{
                    b+=n;
                    s2.push(n);
                }
                n--;
            }
            cout << endl;
            cout << s1.size() << endl;
            while(!s1.empty()){
                cout << s1.top() << " ";
                s1.pop();
            }

            cout << endl;
            cout << s2.size() << endl;
            while(!s2.empty()){
                cout << s2.top() << " ";
                s2.pop();
            }
        }else{
            cout << "NO";
        }


    
}